### PR TITLE
[MAINTENANCE] Update Cloud Tests for release this week + Revert pin on `typing-extension`

### PR DESCRIPTION
**Pull request by Kilo59**
# Fix tests for release
* Reverts #7970, and #7977, which was addressed by an updated `typing-extensions` package 
* 
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -31,6 +31,6 @@ requests>=2.20
 ruamel.yaml>=0.16,<0.17.18  # See https://github.com/great-expectations/great_expectations/issues/4152
 scipy>=1.6.0
 tqdm>=4.59.0
-typing-extensions>=3.10.0.0,<4.6.0 # Leverage type annotations from recent Python releases
+typing-extensions>=3.10.0.0 # Leverage type annotations from recent Python releases
```
Comment (Kilo59) [inline requirements.txt:34]: Removed this pin due to this fix in pydantic and subsequent release
- https://github.com/pydantic/pydantic/pull/5826
- https://github.com/pydantic/pydantic/releases/tag/v1.10.8

File: tests/checkpoint/cloud_config.py [hunk 2]
```diff
@@ -56,11 +56,13 @@ def store_set(self, key, value, **kwargs):
...
     if self.ge_cloud_resource_type == GXCloudRESTResource.EXPECTATION_SUITE:
-        return (
-            GXCloudRESTResource.EXPECTATION_SUITE,
-            "1212e79d-f751-4c6e-921d-26de2b1db174",
-            "suite_name",
-        )
+        return [
+            (
+                GXCloudRESTResource.EXPECTATION_SUITE,
+                "1212e79d-f751-4c6e-921d-26de2b1db174",
```
Comment (alexsherstinsky) [inline tests/checkpoint/cloud_config.py:62]: @Shinnnyshinshin was this key generated by GX-Cloud for the test?  Thanks.

**Review by Kilo59 — COMMENTED** [review]
✅ I approve

**Review by alexsherstinsky — APPROVED** [review]
LGTM

File: tests/checkpoint/cloud_config.py (hunk 2 above)
Comment (Shinnnyshinshin) [reply tests/checkpoint/cloud_config.py:62]: Hi @alexsherstinsky great question. It's actually part of a fixture that is used in a Mock: 

https://github.com/great-expectations/great_expectations/blob/184144211940df099c440b2eb5e9d0c4691ea3ce/tests/checkpoint/test_checkpoint.py#L4799

**Review by Shinnnyshinshin — APPROVED** [review]
✅  😄

**Pull request merged**